File: Bwl.Imaging/Geometry/Structures/RectangleFC.vb
Imports System.Drawing
Imports System.Runtime.Serialization

Public Class RectangleFC
    Implements ICloneable

    Public Property RectangleF As RectangleF

    Public Function Clone() As Object Implements ICloneable.Clone
        Return New RectangleFC(RectangleF)
    End Function

    Public Sub New(rect As RectangleF)
        Me.RectangleF = rect
    End Sub

    Public Sub New(rect As Rectangle)
        Me.RectangleF = rect
    End Sub

    Public Sub New(x1 As Single, y1 As Single, x2 As Single, y2 As Single)
        Me.RectangleF = Drawing.RectangleF.FromLTRB(x1, y1, x2, y2)
    End Sub

    Public Sub New(location As PointF, size As SizeF)
        Me.RectangleF = New RectangleF(location, size)
    End Sub

    Public Sub New()

    End Sub

    Public Overrides Function ToString() As String
        Return "RectangleC: " + RectangleF.ToString
    End Function

End Class

File: Bwl.Imaging/Geometry/Structures/RectangleFC.cs
using System;
using System.Drawing;

namespace Bwl.Imaging
{

    public class RectangleFC : ICloneable
    {

        public RectangleF RectangleF { get; set; }

        public object Clone()
        {
            return new RectangleFC(RectangleF);
        }

        public RectangleFC(RectangleF rect)
        {
            RectangleF = rect;
        }

        public RectangleFC(Rectangle rect)
        {
            RectangleF = rect;
        }

        public RectangleFC(float x1, float y1, float x2, float y2)
        {
            RectangleF = RectangleF.FromLTRB(x1, y1, x2, y2);
        }

        public RectangleFC(PointF location, SizeF size)
        {
            RectangleF = new RectangleF(location, size);
        }

        public RectangleFC()
        {

        }

        public override string ToString()
        {
            return "RectangleC: " + RectangleF.ToString();
        }

    }
}